### PR TITLE
Update docstrings for API commands

### PR DESCRIPTION
**Pull request by billschereriii**
Technical, grammatical, and spelling updates for the docstrings.  A second pass, to make sure the docs build correctly, will follow in a second PR

### CODE REVIEW

**Review by mellis13 — CHANGES_REQUESTED** [review]
Great update to the docs!  

The majority of comments are around prefixing behavior.  For these comments, I tried to comment on each instance to make updates easier.

File: include/addressatcommand.h [hunk 1]
```diff
@@ -72,9 +72,9 @@ class AddressAtCommand : public NonKeyedCommand
         *   \brief Returns host of database node
         *   \param address The address of the database node
         *   \returns The host of the database node
-        *   \throws SRRuntimeException if ':' is at the start of address,
-        *           or if ':' is not found in address, or if allocating
-        *           storage for the host string fails.
+        *   \throw SRRuntimeException if ':' is at the start of address,
+        *          or if ':' is not found in address, or if allocating
+        *          storage for the host string fails.
```
[the remark this answers is not in the document]
Comment (billschereriii) [reply include/addressatcommand.h:77]: Yeah, we at least shouldn't say that we'll throw a RuntimeException in this case

File: include/c_client.h [hunk 2]
```diff
@@ -50,25 +50,32 @@ SRError SmartRedisCClient(bool cluster, void **new_client);
...
 */
 SRError DeleteCClient(void** c_client);
 
 /*!
 *   \brief Put a DataSet object into the database
-*   \param c_client A pointer to c client
-*                   to use for communication
+*   \details The final dataset key under which the dataset is stored
+*            is generated from the name that was supplied when the
+*            dataset was created and may be prefixed. See set_data_source()
```
[the remark this answers is not in the document]
Comment (billschereriii) [reply include/c_client.h:63]: fixed for put_dataset, put_tensor, set_script[_from_file], set_model[_from_file], set_function (only in Python)

File: include/c_client.h [hunk 3]
```diff
@@ -79,13 +86,16 @@ SRError get_dataset(void* c_client,
                     void** dataset);
 
 /*!
-*   \brief Move a DataSet to a new key
-*   \param c_client A pointer to c client to use for communication
-*   \param name The name of the dataset object
-*   \param name_length The length of the name c-string,
+*   \brief Move a DataSet to a new name key
+*   \details The dataset key used to locate the dataset to be renamed
```
[the remark this answers is not in the document]
Comment (billschereriii) [reply include/c_client.h:90]: Fixed for copy/rename for tensor and dataset

File: include/c_client.h [hunk 4]
```diff
@@ -457,17 +536,20 @@ SRError run_script(void* c_client,
 /*!
 *   \brief Run a model in the database using the specificed input and
 *          output tensors
-*   \param c_client A pointer to c client to use for communication
+*   \details The model key that is used to locate the model to be run
+*            may be prefixed. See set_data_source()
```
[the remark this answers is not in the document]
Comment (billschereriii) [reply include/c_client.h:540]: Good catch!

File: include/c_client.h [hunk 5]
```diff
@@ -344,107 +421,135 @@ SRError set_model(void* c_client,
                   const size_t n_outputs);
 
 /*!
-*   \brief Get a model in the databasea
-*   \param c_client A pointer to c client to use for communication
-*   \param key The key to use to get the model
-*   \param key_length The length of the key c-string,
-*                     excluding null terminating character
-*   \param model_length The length of the model buffer c-string,
+*   \brief Get a model in the database.   The memory associated with the
```
Comment (mellis13) [inline include/c_client.h:424]: Extra spaces?

**Review by mellis13 — APPROVED** [review]
Great improvement to docs!

I tried to point out spots where we might want to continue to improve our use of "name" vs "key", since that was a recent push.  Some of these may be subjective, so not all of them necessarily need to be changed.

File: src/python/module/smartredis/client.py [hunk 6]
```diff
@@ -66,265 +66,340 @@ def __init__(self, address=None, cluster=False):
...
+        super().put_tensor(name, dtype, data)
 
     @exception_handler
-    def get_tensor(self, key):
+    def get_tensor(self, name):
         """Get a tensor from the database
 
-        :param key: key to get tensor from
-        :type key: str
+        The tensor name used to locate the tensor
```
[the remark this answers is not in the document]
Comment (billschereriii) [reply src/python/module/smartredis/client.py:91]: Agreed

**Pull request merged**